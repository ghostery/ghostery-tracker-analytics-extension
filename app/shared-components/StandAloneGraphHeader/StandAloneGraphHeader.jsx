import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import Nav from 'react-bootstrap/Nav';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';

import './StandAloneGraphHeader.scss';

const StandAloneGraphHeader = ({
  title,
  children,
  small,
  handleClick,
  tooltipText,
  hideSeeDetails,
  tooltipRef,
  expandable,
  expanded,
  expandGraph,
}) => {
  const standAloneGraphHeaderClasses = ClassNames('StandAloneGraphHeader__mainInfo', {
    'd-flex': small,
    'align-items-center': small,
    'justify-content-between': small,
    small,
  });
  const standAloneGraphHeaderArrows = ClassNames(
    'StandAloneGraphHeader__arrows',
    'd-flex',
    'align-items-center',
    'justify-items-center',
    { StandAloneGraphHeader__collapse: expanded },
  );

  return (
    <div className="d-flex">
      <div className={standAloneGraphHeaderClasses}>
        <div className="StandAloneGraphHeader__header d-flex justify-content-between align-items-center">
          <div className="d-flex align-items-center">
            <h3>{title}</h3>
            <OverlayTrigger
              container={tooltipRef}
              placement="auto"
              overlay={(
                <Tooltip className="tooltipLeft">
                  {tooltipText}
                </Tooltip>
              )}
            >
              <img
                src={[chrome.extension.getURL('dist/images/shared/info-icon.svg')]}
                className="infoIcon"
                width="16"
                height="16"
                alt={`More info hover: ${tooltipText}`}
              />
            </OverlayTrigger>
          </div>
          {!small && expandable && (
            <OverlayTrigger
              container={tooltipRef}
              placement="auto"
              delay={500}
              overlay={(
                <Tooltip className="tooltipLeft">
                  {expanded ? 'Collapse' : 'Expand'}
                </Tooltip>
              )}
            >
              <div className={standAloneGraphHeaderArrows} onClick={expandGraph}>
                {expanded ? (
                  <svg xmlns="http://www.w3.org/2000/svg" width="23" height="9">
                    <path fill="#EBEBEB" d="M9.784 4.982a.626.626 0 0 0-.003-.88L6.004.276c-.452-.463-1.129.231-.677.694L8.17 3.886c.094.097.065.175-.079.175H.85a.485.485 0 0 0-.479.491c0 .272.214.492.48.492H8.09c.138 0 .175.076.079.175L5.327 8.135c-.452.463.225 1.158.677.695l3.78-3.848zm3.071-.86a.626.626 0 0 0 .004.88l3.777 3.828c.451.463 1.129-.231.677-.695L14.47 5.22c-.095-.097-.065-.175.078-.175h7.242c.265 0 .48-.22.48-.492a.485.485 0 0 0-.48-.49h-7.242c-.137 0-.175-.077-.078-.176L17.313.969c.452-.463-.226-1.157-.677-.694l-3.78 3.848z" />
                  </svg>
                ) : (
                  <svg xmlns="http://www.w3.org/2000/svg" width="17" height="8">
                    <path fill="#EBEBEB" d="M15.424 3.556c.114 0 .144-.07.065-.158L13.092.765c-.38-.418.19-1.045.571-.627l3.183 3.456a.594.594 0 0 1 .004.794l-3.187 3.474c-.38.418-.951-.21-.57-.627l2.396-2.633c.08-.09.05-.158-.065-.158H1.577c-.114 0-.145.07-.066.158l2.397 2.633c.38.418-.19 1.045-.571.627L.154 4.406a.594.594 0 0 1-.004-.794L3.337.138c.38-.418.951.21.57.627L1.512 3.398c-.08.09-.05.158.066.158h13.847z" />
                  </svg>
                )}
              </div>
            </OverlayTrigger>
          )}
        </div>
        {children}
      </div>
      {small && !hideSeeDetails && (
        <Nav.Link
          className="StandAloneGraphHeader__seeDetailsLink"
          onClick={handleClick}
        >
          See details
        </Nav.Link>
      )}
    </div>
  );
};

StandAloneGraphHeader.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  small: PropTypes.bool,
  handleClick: PropTypes.func,
  hideSeeDetails: PropTypes.bool,
  tooltipRef: PropTypes.instanceOf(PropTypes.element),
  tooltipText: PropTypes.string.isRequired,
  expandable: PropTypes.bool,
  expanded: PropTypes.bool,
  expandGraph: PropTypes.func,
};

StandAloneGraphHeader.defaultProps = {
  children: null,
  small: false,
  handleClick: null,
  hideSeeDetails: false,
  tooltipRef: null,
  expandable: false,
  expanded: false,
  expandGraph: null,
};

export default StandAloneGraphHeader;
