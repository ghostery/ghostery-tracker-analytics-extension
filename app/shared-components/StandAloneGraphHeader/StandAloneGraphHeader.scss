@import '../../utils/scss/color-variables.scss';
@import '../../utils/scss/z-index-variables.scss';

.StandAloneGraph__background {
  position: relative;
  background-color: $opaque-background;
}

.StandAloneGraph__blueStripe {
  background-color: $ghosty-blue;
  width: 3px;
}

.StandAloneGraphHeader__mainInfo {
  width: 100%;
  height: 70px;

  p {
    margin-left: 20px;
  }

  &.small {
    width: 220px;
    height: 40px;
  }

  &.small > div > div > h3 {
    margin-left: 10px;
    font-size: 11px;
  }
}

.StandAloneGraphHeader__header {
  margin: 10px 0 12px;

  h3 {
    margin: 0 8px 0 20px;
  }
}

.StandAloneGraphHeader__arrows {
  position: relative;
  top: -1px;
  right: 16px;
  width: 17px;
  height: 12px;
  cursor: pointer;

  &.StandAloneGraphHeader__collapse {
    width: 23px;
  }

  &:hover path {
    fill: $ghosty-blue;
  }
}

.StandAloneGraphHeader__keyData {
  font-weight: 700;
  font-size: 14px;
  line-height: 11px;
  color: $ghosty-blue;

  p {
    margin: 0;
    text-align: right;
  }

  &.small {
    margin-right: 10px;
    font-weight: 700;
    text-align: center;
  }
}

#StandAloneGraphHeader__smallSubHeader {
  font-size: 11px;
}

.StandAloneGraphHeader__seeDetailsLink {
  position: absolute;
  z-index: $z-index-see-details-link;
  bottom: 3px;
  right: 5px;
  padding: 0;
  font-weight: 400;
  font-size: 10px;
  color: $white;
}
