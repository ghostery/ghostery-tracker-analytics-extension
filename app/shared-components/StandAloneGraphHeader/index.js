import { connect } from 'react-redux';

import StandAloneGraphHeader from './StandAloneGraphHeader.jsx';

const mapStateToProps = state => ({
  tooltipRef: state.UserInterfaceReducer.tooltipRef,
});

export default connect(mapStateToProps)(StandAloneGraphHeader);
