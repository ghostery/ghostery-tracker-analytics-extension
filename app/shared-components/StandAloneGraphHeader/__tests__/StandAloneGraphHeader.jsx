import React from 'react';
import { shallow } from 'enzyme';
import chrome from 'sinon-chrome';
import path from 'path';
import fakeProps from 'react-fake-props';

import StandAloneGraphHeader from '../StandAloneGraphHeader';

jest.mock('../../../../src/classes/Globals', () => ({
  BROWSER_INFO: { displayName: 'Chrome' },
}));

describe('app/shared-components/StandAloneGraphHeader/', () => {
  describe('Snapshot tests with shallow', () => {
    test('StandAloneGraphHeader component should render', () => {
      global.chrome = chrome;
      const componentPath = path.join(__dirname, '../StandAloneGraphHeader.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<StandAloneGraphHeader {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
