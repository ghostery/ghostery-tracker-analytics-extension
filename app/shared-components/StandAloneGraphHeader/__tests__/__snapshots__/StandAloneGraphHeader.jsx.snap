// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/StandAloneGraphHeader/ Snapshot tests with shallow StandAloneGraphHeader component should render 1`] = `
<div
  className="d-flex"
>
  <div
    className="StandAloneGraphHeader__mainInfo"
  >
    <div
      className="StandAloneGraphHeader__header d-flex justify-content-between align-items-center"
    >
      <div
        className="d-flex align-items-center"
      >
        <h3>
          title
        </h3>
        <OverlayTrigger
          container={null}
          defaultOverlayShown={false}
          overlay={
            <Tooltip
              className="tooltipLeft"
              placement="right"
            >
              tooltipText
            </Tooltip>
          }
          placement="auto"
          trigger={
            Array [
              "hover",
              "focus",
            ]
          }
        >
          <img
            alt="More info hover: tooltipText"
            className="infoIcon"
            height="16"
            src={
              Array [
                undefined,
              ]
            }
            width="16"
          />
        </OverlayTrigger>
      </div>
    </div>
  </div>
</div>
`;
